nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/FetchSlack.java

```diff
@@ -0,0 +1,302 @@
+        session.transfer(requestFlowFile, REL_SUCCESS_REQ);
+      } else {
+        successfulResponses.forEach(session::remove);
+      }
+    } catch (IOException e) {
+      session.transfer(requestFlowFile, REL_FAILURE);
+    }
+  }
+
+  private List<JsonObject> collectFilesToDownload(InputStream requestInputStream) {
+    List<JsonObject> filesToDownload = new ArrayList<>();
+    try (JsonReader reader = Json.createReader(requestInputStream)) {
+      JsonObject jsonObject = reader.readObject();
+      JsonArray files = jsonObject.getJsonArray("files");
+      if (files != null) {
+        for (JsonValue file : files) {
+          String url = ((JsonObject) file).getString("url_private_download");
+          if (url != null) {
+            filesToDownload.add((JsonObject) file);
+          }
+        }
+      }
+    }
+
+    return filesToDownload;
+  }
+
+  private HashMap<String, String> fillAttributes(Map<String, String> attributes, JsonObject file, HttpResponse response) {
+    HashMap<String, String> map = new HashMap<>(attributes);
+    String value = response.getFirstHeader("Content-Disposition").getValue();
+    Matcher matcher = filenamePattern.matcher(value);
+    if (matcher.matches()) {
+      map.put("response.filename", matcher.group(1));
+    }
+
+    fromheader(map, response, "Content-Length", "response.content-length");
+    fromheader(map, response, "Content-type", "response.content-type");
+    fromheader(map, response, "Date", "response.date");
+
+    fromJson(map, file, "name", "slack.file.name");
+    fromJson(map, file, "mimetype", "slack.file.mimetype");
+    fromJson(map, file, "filetype", "slack.file.filetype");
+    fromJson(map, file, "title", "slack.file.title");
+    fromJson(map, file, "id", "slack.file.id");
+    fromJson(map, file, "created", "slack.file.created");
+    fromJson(map, file, "size", "slack.file.size");
+
+    return map;
+  }
+
+  private void fromJson(Map<String, String> attributes, JsonObject file, String jsonKey, String attribute) {
+    Optional.ofNullable(file.getOrDefault(jsonKey, null)).ifPresent(
+      jsonValue -> {
+        ValueType valueType = jsonValue.getValueType();
+        String stringValue = null;
+        if (valueType.equals(ValueType.NUMBER)) {
+          stringValue = String.valueOf(file.getInt(jsonKey));
+        } else if (valueType.equals(ValueType.STRING)) {
+          stringValue = file.getString(jsonKey);
+        }
+
+        if (stringValue != null) {
+          attributes.put(attribute, stringValue);
+        }
+      }
+    );
+
+  }
+
+  private void fromheader(Map<String, String> attributes, HttpResponse response, String header, String attribute) {
+    Header firstHeader = response.getFirstHeader(header);
+    if (firstHeader != null)
+      attributes.put(attribute, firstHeader.getValue());
+  }
+
+  private boolean inStatusCodeFamily(int family, int statusCode) {
+    return statusCode >= family && statusCode < family + 100;
```

Comment:
Should this have a boundary check to ensure the `family` is divisible by 100? Otherwise a family of 499 could be provided, and a status code of 501 would return true.

Reply:
As this is a private helper method that is called only with constant numbers, I don't see the need for boundary check